# EE283: Advanced Bioinformatics

In this class a combinaiton of bash (HPC), python and R programming has been used. 

[Week 1](homework1.md): **Basic Bash Commands** 

[Week 2](homework2.md): **Creating Symlinks to RNAseq, DNAseq and ATACseq data**

[Week 3](week3.md): **Mapping**

  - RNAseq: HISAT2 and samtools was used to map RNA data. Array work was applied.
  - DNAseq and ATACseq: bwa and samtools was used to map both DNA and ATAC sequencing data. Array work was applied.

[Week 4](week4.md): **Visualizing Mapping Coverage**  

 For this assignment deeptools was used and the data was visualized both on IGV and plots.

[Week 5](week5.md): **SNP Calling**

 GATK software was used to call SNPs and heatmaps (made with python seaborn package) were generated to visualize the data.

[Week 7](week7.md): **Differential Gene Expression Analysis of RNAseq data**

 Programs/Packages used:
   - FeatureCount on HPC.
   - R programming DESeq2 and ggplot 
   - Python pyDESeq2, pandas, seaborn, numpy, matplotlib 

 Data visulization:
    PCA plot, volcano plot, Ma plot, histogram, and heatmaps were generated to demonstrate the data.

[Week 8](week8.md): **R programming**

[Week 9](week9.md): **Figure Generation**

[Week 10](week10.md): **Epitranscriptomic Analysis of long-read RNA sample**

Programs/Packages used:
 - Dorado: ONT basecaller ran on HPC 
 - Minimap2: used to map ONT RNAseq data on HPC3 
 - Modkit: Modification caller ran on hpc. 
 - R ggplot package to visualize data. 
